### PR TITLE
Print all JIT instrumentation in EXPLAIN ANALYZE VERBOSE mode

### PR DESCRIPTION
**Pull request by QingMa-gp**
For example:
```
postgres=# explain analyze verbose select * from test limit 10;
                                                           QUERY PLAN                                                            
---------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=35.50..35.67 rows=10 width=4) (actual time=16.941..16.944 rows=0 loops=1)
   Output: c1
   ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=35.50..36.01 rows=30 width=4) (actual time=15.436..15.438 rows=0 loops=1)
         Output: c1
         ->  Limit  (cost=35.50..35.61 rows=10 width=4) (actual time=0.000..1.317 rows=0 loops=1)
               Output: c1
               ->  Seq Scan on public.test  (cost=0.00..355.00 rows=32100 width=4) (actual time=0.000..0.029 rows=0 loops=1)
                     Output: c1
 Optimizer: Postgres query optimizer
 Settings: jit = 'on', jit_above_cost = '0'
 Planning Time: 2.616 ms
   (slice0)    Executor memory: 37K bytes.
   (slice1)    Executor memory: 38K bytes avg x 3 workers, 38K bytes max (seg0).
 Memory used:  128000kB
 JIT:
   Options: Inlining false, Optimization false, Expressions true, Deforming true.
   (slice0): Functions: 2.00. Timing: 1.535 ms total.
    seg-1: Function 2, Generation 0.117 ms, Inlining 0.000 ms, Optimization 0.112 ms, Emission 1.306 ms, Total 1.535 ms.
   (slice1): Functions: 1.00 avg x 3 workers, 1.00 max (seg0). Timing: 1.268 ms avg x 3 workers, 1.312 ms max (seg2).
    seg0: Function 1, Generation 0.078 ms, Inlining 0.000 ms, Optimization 0.089 ms, Emission 1.081 ms, Total 1.248 ms.
    seg1: Function 1, Generation 0.079 ms, Inlining 0.000 ms, Optimization 0.085 ms, Emission 1.081 ms, Total 1.245 ms.
    seg2: Function 1, Generation 0.079 ms, Inlining 0.000 ms, Optimization 0.089 ms, Emission 1.143 ms, Total 1.312 ms.
 Execution Time: 29.889 ms
(23 rows)
```

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by kainwen — APPROVED** [review]
LGTM.

The name `seg_functions_generation_inlining_optimization_emission` is a little longer, lets see if others have good advice.

**Comment by QingMa-gp** [comment]
> > seg_functions_generation_inlining_optimization_emission/0_1_0.105 ms_0.000 ms_0.470 ms_2.926 ms/1_1_0.240 ms_0.000 ms_0.217 ms_2.606 ms/2_1_0.162 ms_0.000 ms_0.189 ms_2.453 ms//end.
> 
> This is very hard to read.
> 
> Why not just do what upstream does in `ExplainPrintJIT`:
> 
> ```
> appendStringInfo(es->str,
> 				 "Timing: %s %.3f ms, %s %.3f ms, %s %.3f ms, %s %.3f ms, %s %.3f ms\n",
> 				 "Generation", 1000.0 * INSTR_TIME_GET_DOUBLE(ji->generation_counter),
> 				 "Inlining", 1000.0 * INSTR_TIME_GET_DOUBLE(ji->inlining_counter),
> 				 "Optimization", 1000.0 * INSTR_TIME_GET_DOUBLE(ji->optimization_counter),
> 				 "Emission", 1000.0 * INSTR_TIME_GET_DOUBLE(ji->emission_counter),
> 				 "Total", 1000.0 * INSTR_TIME_GET_DOUBLE(total_time));
> ```

Output will be very long when there are many segments and slices. Therefore, we referred to `gp_enable_explain_allstat` and adopted this approach.

**Review by Tao-T — APPROVED** [review]
LGTM

**Comment by soumyadeep2007** [comment]
> Output will be very long when there are many segments and slices. Therefore, we referred to gp_enable_explain_allstat and adopted this format. https://github.com/greenplum-db/gpdb/blob/main/src/backend/commands/explain_gp.c#L1734-L1746

This would create very long lines in the output. Even if it is long (this is VERBOSE mode which is fine), its better to have the output separated into newlines.

```
seg-1: Functions: 2 Generation: 0.198 ms Inlining: 0.000 ms Optimization: 0.234 ms Emission: 1.828 ms
seg0: Functions: ...
...
```

**Comment by QingMa-gp** [comment]
> > Output will be very long when there are many segments and slices. Therefore, we referred to gp_enable_explain_allstat and adopted this format. https://github.com/greenplum-db/gpdb/blob/main/src/backend/commands/explain_gp.c#L1734-L1746
> 
> This would create very long lines in the output. Since, this is VERBOSE mode, its okay to have the output separated into newlines. Much better to have users scroll vertically than horizontally.
> 
> ```
> seg-1: Functions: 2 Generation: 0.198 ms Inlining: 0.000 ms Optimization: 0.234 ms Emission: 1.828 ms
> seg0: Functions: ...
> ...
> ```

Make sense, it has been modified. Thanks, @soumyadeep2007.

File: src/backend/commands/explain_gp.c [hunk 1]
```diff
@@ -2327,17 +2328,16 @@ cdbexplain_printJITSummary(ExplainState *es, QueryDesc *queryDesc)
...
-						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->generation_counter),
-						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->inlining_counter),
-						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->optimization_counter),
-						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->emission_counter),
-						 1000.0 * INSTR_TIME_GET_DOUBLE(total_time));
+			appendStringInfoSpaces(allstats, es->indent * 2 + 1);
+			appendStringInfo(allstats,
+							 "%s%d: %s %zu, %s %.3f ms, %s %.3f ms, %s %.3f ms, %s %.3f ms, %s %.3f ms.\n",
+							 "seg", ss->segindex0 +j,
+							 "Function", ji->created_functions,
```
Comment (soumyadeep2007) [inline src/backend/commands/explain_gp.c:2335]: Nit: Function/Functions

**Review by soumyadeep2007 — APPROVED** [review]
LGTM! Looks much better, thanks!

File: src/backend/commands/explain_gp.c (hunk 1 above)
Comment (QingMa-gp) [reply src/backend/commands/explain_gp.c]: Done.

**Comment by QingMa-gp** [comment]
Thanks for the review.

**Pull request merged**